src/compiler/checker.ts

```diff
@@ -46,6 +46,7 @@ namespace ts {
         const compilerOptions = host.getCompilerOptions();
         const languageVersion = compilerOptions.target || ScriptTarget.ES3;
         const modulekind = compilerOptions.module ? compilerOptions.module : languageVersion === ScriptTarget.ES6 ? ModuleKind.ES6 : ModuleKind.None;
+        const allowSyntheticDefaultImports = compilerOptions.hasOwnProperty("allowSyntheticDefaultImports") ? compilerOptions.allowSyntheticDefaultImports : modulekind === ModuleKind.System;

```

Comment:
how about `compilerOptions.allowSyntheticDefaultImports !== undefined` or `typeof compilerOptions.allowSyntheticDefaultImports !== "undefined"`?

i am just concerned about readability, and refactor/rename support in tooling.